src/diagnostics/core_diagnostics.jl

```diff
@@ -16,19 +27,160 @@ add_diagnostic_variable!(
+    comments = "Northward (meridional) wind component, a prognostic variable",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        return copy(Geometry.VVector.(cache.ᶜu))
+    end,
+)
+
+###
+# W velocity (3d)
+###
+# TODO: may want to convert to omega (Lagrangian pressure tendency) as standard output,
+# but this is probably more useful for now
+add_diagnostic_variable!(
+    short_name = "vertical_wind",
+    long_name = "Vertical Wind",
+    units = "m s^-1",
+    comments = "Vertical wind component, a prognostic variable",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        return copy(Geometry.WVector.(cache.ᶜu))
+    end,
+)
+
+###
+# Temperature (3d)
+###
+add_diagnostic_variable!(
+    short_name = "air_temperature",
+    long_name = "Air Temperature",
+    units = "K",
+    comments = "Temperature of air",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        thermo_params = CAP.thermodynamics_params(cache.params)
+        return copy(TD.air_temperature.(thermo_params, cache.ᶜts))
+    end,
+)
+
+###
+# Potential temperature (3d)
+###
+add_diagnostic_variable!(
+    short_name = "air_potential_temperature",
+    long_name = "Air potential temperature",
+    units = "K",
+    comments = "Potential temperature of air",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        thermo_params = CAP.thermodynamics_params(cache.params)
+        return copy(TD.dry_pottemp.(thermo_params, cache.ᶜts))
+    end,
+)
+
+###
+# Air pressure (3d)
+###
+add_diagnostic_variable!(
+    short_name = "air_pressure",
+    long_name = "Air pressure",
+    units = "Pa",
+    comments = "Pressure of air",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        return copy(cache.ᶜp)
+    end,
+)
+
+###
+# Vorticity (3d)
+###
+add_diagnostic_variable!(
+    short_name = "atmosphere_relative_vorticity",
+    long_name = "Vertical component of relative vorticity",
+    units = "s^-1",
+    comments = "Vertical component of relative vorticity",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        ᶜvort = @. Geometry.WVector(curlₕ(state.c.uₕ))
+        if cache.do_dss
+            Spaces.weighted_dss!(ᶜvort)
+        end
```

Comment:
It would be nice to have a comment explaining why we have this if-else

Reply:
`do_dss` is probably dependent on the discretization, but maybe @sriharshakandala or @dennisYatunin will know more.

Reply:
The `do_dss` checks if the underlying `hspace` is a spectral element space. https://github.com/CliMA/ClimaAtmos.jl/blob/main/src/cache/cache.jl#L64
It is not immediately clear to me why a dss is needed here. May be @dennisYatunin knows!

Reply:
Not sure, but my guess is that `curlₕ.(Y.c.uₕ)` generates a Field that is discontinuous across element boundaries (unless we're in a column, in which case `do_dss` is false), and we only want to use continuous Fields for diagnostics.

Reply:
Can `weighted_dss!` automatically recognize if we are in a space where dss makes sense or not?

Reply:
Not currently. We could move this if-statement into `weighted_dss!` in ClimaCore, thought I feel like that would be hiding a bit too much under the hood. Since DSS only makes sense on a spectral element discretization, it would be confusing to see a call to `weighted_dss!` on a column space, even if it gets reduced to a no-op in ClimaCore.

Reply:
I think there's an opportunity here to make things more clear. If one doesn't know what DSS is, this code is completely obscure. If one knows what DSS is, the intent of the code is still obscure. Maybe `weighted_dss!` could be renamed to `enforce_continuity!` ? 

I don't know what's the best way.

Reply:
You could definitely make an `enforce_continuity!` function in ClimaAtmos that dispatches based on the type of the space, if you want to hide this if-statement. Renaming this in ClimaCore might not be such a good idea, since `enforce_continuity!` is less specific than `weighted_dss!` and could lead to ambiguity in the future. @sriharshakandala, thoughts on renaming `weighted_dss!` in ClimaCore?

Reply:
In this case, a name like `smoothen` might make sense as this is what is being done here! I believe we already made a dss a no-op for non-spectral element fields in ClimaCore!

Reply:
However, at other locations, `weighted_dss` makes more sense! We need to rename `dss_op` to `weighted_dss_op` since the underlying operation is weighted dss (i.e. M_-1 * dss(field)) and not simple dss.

Reply:
Ah, I didn't know dss can be a no-op now. In that case, the only use of `do_dss` in ClimaAtmos should be to determine whether or not to allocate a cache for dss, and you can probably remove the if-statement here. As for the name change, maybe we could run that by the rest of the group first; I'm sure people will have lots of opinions!